cmd/query/app/http_handler.go

```diff
@@ -138,11 +138,9 @@ func (aH *APIHandler) handleFunc(
 	if aH.tenancyMgr.Enabled {
 		handler = tenancy.ExtractTenantHTTPHandler(aH.tenancyMgr, handler)
 	}
-	traceMiddleware := otelhttp.NewHandler(
-		otelhttp.WithRouteTag(route, traceResponseHandler(handler)),
-		route,
-		otelhttp.WithTracerProvider(aH.tracer))
-	return router.HandleFunc(route, traceMiddleware.ServeHTTP)
+	handler = traceResponseHandler(handler)
+	handler = otelhttp.WithRouteTag(route, handler)
+	return router.HandleFunc(route, handler.ServeHTTP)
```

Comment:
@yurishkuro is it okay to remove the use of the otelhttp handler here? this removes the double instrumentation problem we were seeing earlier

Reply:
Yes ok to remove trace handler. Does the root tag handler allow you to avoid calling span.setname?

Reply:
@yurishkuro I'm not sure if its because of this handler but I don't see the high cardinality spans anymore. 

<img width="1462" alt="Screenshot 2024-10-26 at 7 35 39 PM" src="https://github.com/user-attachments/assets/aa7df0b6-c8c7-4edc-a975-e54113f61e6b">